code/game/objects/effects/effect_system.dm

```diff
@@ -42,8 +42,7 @@ would spawn and follow the beaker, even if it is carried or thrown.
 /obj/effect/proc/delete()
 	loc = null
 	if(reagents)
-		reagents.my_atom = null
-		reagents.delete()
+		qdel(reagents)

```

Comment:
I mean replacing _everything_ in this proc with qdel(src), since the Destroys will do everything this proc does and more. As it is, this looks like it's just meant to be a ghetto qdel.